homeassistant/components/usb/__init__.py

```diff
@@ -243,26 +248,56 @@ def async_stop(self, event: Event) -> None:
         if self._request_debouncer:
             self._request_debouncer.async_shutdown()
 
-    async def _async_start_monitor(self) -> None:
-        """Start monitoring hardware with pyudev."""
-        if not sys.platform.startswith("linux"):
-            return
+    async def _async_supports_monitoring(self) -> bool:
         info = await system_info.async_get_system_info(self.hass)
         if info.get("docker"):
-            return
+            return False
+
+        return True
+
+    async def _async_start_monitor(self) -> None:
+        """Start monitoring hardware."""
+        if not await self._async_start_monitor_udev():
+            # We fall back to polling to make development possible, this is not a proper
+            # way to run Home Assistant
+            await self._async_start_monitor_polling()
+
+    async def _async_start_monitor_polling(self) -> None:
+        """Start monitoring hardware with polling (for development only!)."""
+        _LOGGER.info(
+            "Falling back to USB port polling for development, libudev is not preset"
+        )
```

Comment:
since the check for udev is not in this method, can we move this log out of it and to the place where it is called instead?

Reply:
As far as I can tell, the *only* installation environment where pyudev is actually used is virtualenv. If a Docker environment is detected, we assume that Supervisor will call `usb/scan` to trigger scans.

Do you think a warning instead of `info` is better?

Reply:
What I meant is the check for `libudev is not preset` is not done inside this method, but on the one that calls it, so my suggestion was to move the log to the place where this method is called. This is just to avoid the log getting outdated if later this method is used for some other reason, and also to keep the log closer to the code it is "describing"

Reply:
Ah, now I understand. Thank you. I think I've addressed this and made the message a little clearer too.